tests/cedict-service/cedict.test.js

```diff
@@ -0,0 +1,279 @@
+/* eslint-env jest */
+import Cedict from '@lexisCs/cedict-service/cedict'
+import CedictPermanentStorage from '@lexisCs/cedict-service/cedict-permanent-storage.js'
+import CedictConfig from '@lexisCs/configurations/cedict.js'
+import CedictC1 from '@lexisCsCedict/cedict-v20191029-c001.json'
+import CedictC2 from '@lexisCsCedict/cedict-v20191029-c002.json'
+import CedictC3 from '@lexisCsCedict/cedict-v20191029-c003.json'
+import CedictC4 from '@lexisCsCedict/cedict-v20191029-c004.json'
+require('fake-indexeddb/auto')
+
+describe('CedictPermanentStorage class', () => {
+  /**
+   * Replaces an original loader with a stub that will load local test data.
+   *
+   * @param {string} url - A URL of a chunk to be fetched.
+   * @returns {Promise<object>} Returns a promise that is resolved with
+   *          an object containing dictionary data.
+   */
+  Cedict.prototype._loadJson = (url) => {
+    const resources = {
+      'http://data-dev.alpheios.net/cedict/cedict-v20191029-c001.json': CedictC1,
+      'http://data-dev.alpheios.net/cedict/cedict-v20191029-c002.json': CedictC2,
+      'http://data-dev.alpheios.net/cedict/cedict-v20191029-c003.json': CedictC3,
+      'http://data-dev.alpheios.net/cedict/cedict-v20191029-c004.json': CedictC4
```

Comment:
just wondering how we might make this less specific to a single version of the database, because then the test would need to be updated with each config change right? Maybe we could just rely on the naming convention and map a url ending in /c001 -> CeditC1 and so on?

Reply:
Agree, that makes sense. I will rename those files. We, however, still need to have database version in a configuration file: it is part of business logic and we need to have it as part of tests. But there is no need to update it with each database update as it is just a number in context of testing.